src/components/ListItem.jsx

```diff
@@ -29,10 +31,17 @@ export function ListItem({
 
 	const handleCheck = async () => {
 		if (!isChecked) {
-			await updateItem(listId, itemId, {
-				dateLastPurchased: new Date(),
-				totalPurchases: totalPurchases + 1,
-			});
+			await updateItem(
+				listId,
+				itemId,
+				dateLastPurchased,
+				dateNextPurchased,
+				dateCreated,
+				{
+					dateLastPurchased: new Date(),
+					totalPurchases: totalPurchases + 1,
+				},
```

Comment:
**Please, there's NO NEED to change this at all, this is code from a previous iteration!** 

But here's a question for you ;)

Are you thinking about using `updateItem` in any other context (other place) than here? If the answer is "NO", then, consider delegating the responsibility of doing

```js
{
  dateLastPurchsed: new Date(),
  totalPurchases: totalPurchases + 1
}
```

To the `updateItem` function instead and hide these lines away from the code that uses the `updateItem` function.

It is not likely that this will happen in TCL, but imagine that you're on a bigger team, and someone else is using this function:

1. Will it ever be the case where `dateLastPurchased` value is something different than `new Date()`?
2. Will it ever be the case where you're increasing the `totalPurchases` by more than one?

If the answer is "NO" or "YES...but probably in the future" to these two questions, then, the "consumer" of this function shouldn't be taking care of this. If the answer is "YES" then leave it as is!

I hope this makes sense, I'm happy to discuss this on Slack if you have questions!

Reply:
That makes sense. I was thinking of refactoring this part, but I didn't since it was the last team's work.